cmd/minikube/cmd/root.go

```diff
@@ -128,9 +129,62 @@ func setFlagsUsingViper() {
 func init() {
 	RootCmd.PersistentFlags().StringP(config.MachineProfile, "p", constants.DefaultMachineName, `The name of the minikube VM being used. This can be set to allow having multiple instances of minikube independently.`)
 	RootCmd.PersistentFlags().StringP(configCmd.Bootstrapper, "b", constants.DefaultClusterBootstrapper, "The name of the cluster bootstrapper that will set up the kubernetes cluster.")
-	RootCmd.AddCommand(configCmd.ConfigCmd)
-	RootCmd.AddCommand(configCmd.AddonsCmd)
-	RootCmd.AddCommand(configCmd.ProfileCmd)
+
+	groups := templates.CommandGroups{
+		{
+			Message: "Basic Commands",
+			Commands: []*cobra.Command{
+				startCmd,
+				statusCmd,
+				stopCmd,
+				deleteCmd,
+				dashboardCmd,
+			},
+		},
+		{
+			Message: "Images Commands",
+			Commands: []*cobra.Command{
+				dockerEnvCmd,
+				cacheCmd,
+			},
+		},
+
+		{
+			Message: "Usability Commands",
+			Commands: []*cobra.Command{
+				serviceCmd,
+				configCmd.AddonsCmd,
+				configCmd.ConfigCmd,
+				configCmd.ProfileCmd,
+			},
+		},
+		{
+			Message: "Advanced Commands:",
```

Comment:
Let's keep the naming consistent. Either have colons everywhere or remove this one.